src/CarbonAware.CLI/src/Commands/Emissions/EmissionsCommand.cs

```diff
@@ -41,17 +70,60 @@ private async Task Run(InvocationContext context)
         var locations = context.ParseResult.GetValueForOption<string[]>(_requiredLocation);
         var startTime = context.ParseResult.GetValueForOption<DateTimeOffset?>(_startTime);
         var endTime = context.ParseResult.GetValueForOption<DateTimeOffset?>(_endTime);
-        
-        var parameters = new CarbonAwareParametersBaseDTO() { 
-            MultipleLocations = locations,
+        var best = context.ParseResult.GetValueForOption<bool>(_best);
+        var average = context.ParseResult.GetValueForOption<bool>(_average);
+
+        var parameters = new CarbonAwareParametersBaseDTO()
+        {
             Start = startTime,
             End = endTime
         };
-
         // Call the aggregator.
-        var results = await aggregator.GetEmissionsDataAsync(parameters);
+        if (best)
+        {
+            parameters.MultipleLocations = locations;
+
+            var result = await aggregator.GetBestEmissionsDataAsync(parameters);
+           
+            if(result != null)
+            {
+                var serializedOuput = JsonSerializer.Serialize((EmissionsDataDTO)result);
+                context.Console.WriteLine(serializedOuput);
+            }
+        }
+        else if (average) 
+        {
+            List<EmissionsDataDTO> emissions = new();
 
-        context.Console.WriteLine(JsonSerializer.Serialize(results));
+            foreach (var location in locations!)
```

Comment:
🐑 can't we do the nullity check for locations up top? We use it in every single case, and ideally they're all assuming it's not null. It's just that the error would be thrown in different places no?

Reply:
Since 'location' is specified as a required option, the inbuilt parser would validate it upfront. So, we can safely assume at this point in the code that the location is not null.